### PR TITLE
Fix EnvGate issues

### LINKED ISSUE #3768: EnvGate issues
I've noticed few inaccuracies with EnvGate.

- using the ```{}.play``` for testing EnvGate-containing synth graphs yields inconsistent behavior, since ```{}.play``` creates ```gate``` and ```fadeTime``` named controls. In case of the fadeTime, this causes the release time to be improper (the ```{}.play```'s default, instead of the provided value), while the attack time works as intended.
```
a = { LPF.ar(Saw.ar(200), 600) * EnvGate.new(fadeTime: 3) }.play; //3s fade-in works
s.queryAllNodes(true); //fadeTime control (created by {}.play) has a value of ~0.02
a.release; //releases in 0.02, instead of 3s
```
This maybe is not a big deal since using SynthDef class is better, but as I was testing things it took me a while to figure out what's going on. If there's no way to prevent ```{}.play``` from creating these ```NamedControl```s, maybe we should make a note about this in the examples.

- Unrelatedly, `i_level` parameter is described as the initial level in the help, but in the implementation it is used as `levelScale` parameter in the EnvGen, scaling the whole Env effectively. If this is the desired behavior, documentation should probably be updated... but I think the ability to set starting value directly would be desirable.

- the actual starting level is defined as ```startVal = (synthFadeTime <= 0);``` so it is 0 for positive fadeTimes and 1 for negative fadeTimes; again, not the behavior documented. Is this the way it should work?

### PR DESCRIPTION
**Pull request by telephon**
@telephon
class library: EnvGate uses direct start level

Instead of relying on an indirect clue, we follow the documentation for
`i_level`, interpreting it as initial level. This fixes #3768,  #3795.

testing: all uses of EnvGate that used the i_level parameter to scale
the resulting envelope. In class library, there is no such use.

this is a change to the implemented, not however to the documented API.

### CODE REVIEW

File: testsuite/classlibrary/TestGraphBuilder.sc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
+TestEnvGate : UnitTest {
+
+	test_number_as_fadeTime {
+
+		var error;
+		try {
+			{ EnvGate(fadeTime:1) }.asSynthDef
+		} { |e| error = e };
+		error.postln;
```
Comment (snappizz) [inline testsuite/classlibrary/TestGraphBuilder.sc]: extraneous postln

Comment (telephon) [reply testsuite/classlibrary/TestGraphBuilder.sc]: thanks, fixed.

File: testsuite/classlibrary/TestGraphBuilder.sc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+TestEnvGate : UnitTest {
+
+	test_number_as_fadeTime {
+
+		var error;
+		try {
+			{ EnvGate(fadeTime:1) }.asSynthDef
+		} { |e| error = e };
+		this.assert(error.isException.not, "number should be a valid EnvGate fadeTime")
```
Comment (brianlheim) [inline testsuite/classlibrary/TestGraphBuilder.sc]: We need an `assertThrows`

Comment (telephon) [reply testsuite/classlibrary/TestGraphBuilder.sc]: OK, I'll implement one.

Comment (telephon) [reply testsuite/classlibrary/TestGraphBuilder.sc]: #3801

Comment (brianlheim) [reply testsuite/classlibrary/TestGraphBuilder.sc]: you are awesome

**Comment by telephon** [comment]
Actually, I'd prefer to merge this into 3.10 in the current form and improve the test later, once the `assertException` pull request has been merged.

**Comment by telephon** [comment]
ah, I'm dizzy. thanks.

**Comment by telephon** [comment]
@brianlheim ok, done!

**Comment by brianlheim** [comment]
Can you please rebase it on current develop? There are merge conflicts (I think this is Githib's issue actually)

File: server/supernova/sc/sc_plugin_interface.cpp [hunk 3]
```diff
@@ -987,6 +987,7 @@ void sc_plugin_interface::buffer_alloc_read_channels(uint32_t index, const char
                                                      uint32_t frames, uint32_t channel_count,
                                                      const uint32_t * channel_data)
 {
+
```
[the remark this answers is not in the document]
Comment (brianlheim) [reply server/supernova/sc/sc_plugin_interface.cpp]: Haha where did this come from? Weird..

File: testsuite/classlibrary/TestGraphBuilder.sc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+
+TestEnvGate : UnitTest {
```
Comment (brianlheim) [inline testsuite/classlibrary/TestGraphBuilder.sc]: This file should be named TestEnvGate; if a more specific name is desired you could use TestEnvGate_GraphBuilding

Comment (telephon) [reply testsuite/classlibrary/TestGraphBuilder.sc]: done, thanks!

File: server/supernova/sc/sc_plugin_interface.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (telephon) [reply server/supernova/sc/sc_plugin_interface.cpp]: maybe from fixing the merge conflicts in github.

**Comment by adcxyz** [comment]
anything still missing here? 
maybe remove the extra blank line ;-), and then its ready?

**Pull request merged**